hypertension/dc-7101-algorithms/py-root/test/utils_test.py

```diff
@@ -1,23 +1,389 @@
+            },
+        ),
+        # 2 readings, but no reading within 30 days
+        (
+            {
+                "bp": [
+                    {
+                        "diastolic": 115,
+                        "systolic": 180,
+                        "date": "2021-10-01"
+                    },
+                    {
+                        "diastolic": 110,
+                        "systolic": 200,
+                        "date": "2021-09-01"
+                    }
+                ],
+                "medication": [],
+                'date_of_claim': '2021-11-09',
+            },
+            {
+                "success": False,
+            },
+        ),
+        # 2 readings, but no second reading within 180 days
+        (
+            {
+                "bp": [
+                    {
+                        "diastolic": 115,
+                        "systolic": 180,
+                        "date": "2021-04-01"
+                    },
+                    {
+                        "diastolic": 110,
+                        "systolic": 200,
+                        "date": "2021-10-10"
+                    }
+                ],
+                "medication": [],
+                'date_of_claim': '2021-11-09',
+            },
+            {
+                "success": False,
+            },
+        ),
+        # 1 reading
+        (
+            {
+                "bp": [
+                    {
+                        "diastolic": 110,
+                        "systolic": 200,
+                        "date": "2021-10-10"
+                    }
+                ],
+                "medication": [],
+                'date_of_claim': '2021-11-09',
+            },
+            {
+                "success": False,
+            },
+        ),
+        # 0 readings
+        (
+            {
+                "bp": [
+                ],
+                "medication": [],
+                'date_of_claim': '2021-11-09',
+            },
+            {
+                "success": False,
+            }
+        ),
+    ],
+)
+def test_sufficient_to_autopopulate(request_data, predominance_calculation):
+    assert sufficient_to_autopopulate(request_data) == predominance_calculation
 
-# TODO: Completely change/update/delete this test to make it better.
-def test_sufficient_to_autopopulate():
-    delete_or_replace_this_test_data_when_you_write_this_test_properly = [
-        {
-            "diastolic": 110,
-            "systolic": 200,
-            "date": "2021-11-01",
-            "date_object": datetime.fromisoformat("2021-11-01")
-        },
-        {
-            "diastolic": 110,
-            "systolic": 200,
-            "date": "2021-09-01",
-            "date_object": datetime.fromisoformat("2021-09-01")
-        }
-    ]
-
-    assert sufficient_to_autopopulate(data_sufficient_to_autopopulate["bp"]) == delete_or_replace_this_test_data_when_you_write_this_test_properly
+@pytest.mark.parametrize(
```

Comment:
we might not need all the test cases for this function, but I thought it wouldn't hurt to be more thorough

Reply:
They look good to me!

(Resolved)